File: ezSCORES/ezSCORES.Model/Matches.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ezSCORES.Model
{
	public class Matches
	{
		public int Id { get; set; }

		public int FixtureId { get; set; }

		public int HomeTeamId { get; set; }

		public int AwayTeamId { get; set; }

		public int StadiumId { get; set; }

		public int? WinnerId { get; set; }

		public DateTime DateAndTime { get; set; }

		public bool IsCompleted { get; set; }

		public bool IsCompletedInRegullarTime { get; set; }
		public bool IsUnderway { get; set; }
		public virtual CompetitionsTeams AwayTeam { get; set; } = null!;

		public virtual ICollection<CompetitionsRefereesMatches> CompetitionsRefereesMatches { get; set; } = new List<CompetitionsRefereesMatches>();

		public virtual ICollection<Goals> Goals { get; set; } = new List<Goals>();

		public virtual CompetitionsTeams HomeTeam { get; set; } = null!;

		public virtual Stadiums Stadium { get; set; } = null!;
	}
}

File: ezSCORES/UI/ezscores_desktop/lib/models/matches.dart
import 'package:ezscores_desktop/models/competitionsRefereesMatches.dart';
import 'package:ezscores_desktop/models/competitionsTeams.dart';
import 'package:ezscores_desktop/models/goals.dart';
import 'package:ezscores_desktop/models/stadiums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'matches.g.dart';

@JsonSerializable()
class Matches {
	int? id;
	int? fixtureId;
	int? homeTeamId;
	int? awayTeamId;
	int? stadiumId;
	int? winnerId;
	DateTime? dateAndTime;
	bool? isCompleted;
	bool? isCompletedInRegullarTime;
	bool? isUnderway;
  CompetitionsTeams? homeTeam;
  CompetitionsTeams? awayTeam;
  List<CompetitionsRefereesMatches>? competitionsRefereesMatches;
  List<Goals>? goals;
  Stadiums? stadium;

	Matches();

	factory Matches.fromJson(Map<String, dynamic> json) => _$MatchesFromJson(json);

	Map<String, dynamic> toJson() => _$MatchesToJson(this);
}
